tests/conftest.py

```diff
@@ -0,0 +1,132 @@
+"""igvm - Shared functions
+
+Copyright (c) 2020 InnoGames GmbH
+"""
+from datetime import datetime, timezone
+from math import ceil, log
+from re import match
+from shlex import quote
+
+from adminapi.dataset import Query
+from adminapi.exceptions import DatasetError
+from adminapi.filters import Any, Regexp
+from libvirt import VIR_DOMAIN_RUNNING
+
+from igvm.hypervisor import Hypervisor
+from igvm.settings import HYPERVISOR_ATTRIBUTES
+from tests import (
+    IGVM_LOCKED_TIMEOUT,
+    JENKINS_EXECUTOR,
+    PYTEST_XDIST_WORKER,
+    PYTEST_XDIST_WORKER_COUNT,
+    VM_HOSTNAME_PATTERN,
+    VM_NET,
+)
+
+
+def cmd(cmd, *args, **kwargs):
+    escaped_args = [quote(str(arg)) for arg in args]
+
+    escaped_kwargs = {}
+    for key, value in kwargs.items():
+        escaped_kwargs[key] = quote(str(value))
+
+    return cmd.format(*escaped_args, **escaped_kwargs)
+
+
+def clean_all(route_network, vm_hostname=None):
+    # Cancelled builds are forcefully killed by Jenkins. They did not have the
+    # opportunity to clean up so we forcibly destroy everything found on any HV
+    # which would interrupt our work in the current JENKINS_EXECUTOR.
+    hvs = [Hypervisor(o) for o in Query({
+        'servertype': 'hypervisor',
+        'environment': 'testing',
+        'vlan_networks': route_network,
+        'state': 'online',
+    }, HYPERVISOR_ATTRIBUTES)]
+
+    # If a VM hostname is given, only that will be cleaned from HVs.
+    if vm_hostname is None:
+        pattern = '^([0-9]+_)?{}$'.format(
+            VM_HOSTNAME_PATTERN.format(JENKINS_EXECUTOR, '[0-9]+'),
+        )
+    else:
+        pattern = '^([0-9]+_)?{}$'.format(vm_hostname)
+
+    # Clean HVs one by one.
+    for hv in hvs:
+        clean_hv(hv, pattern)
```

Comment:
If two people run manual tests at the same time, wouldn't one cleanup the other, since the executor would be the same (manual) and the pattern is matching any numbers?

Reply:
Ok, thinking again, the idea is not to have two people running tests at once, but just parallelizing one testrun, right?

Reply:
it was never possible to run more than one manual test at the same time. so this changes nothing in that regard. but yes they would clean up each other if that happens :D but it would also have happened before.
but yes it is fully intentional that tests clean up everything that was left